src/app-layer-smtp.c

```diff
@@ -312,71 +307,48 @@ static void SMTPConfigure(void) {
         if (extract_urls_schemes) {
             ConfNode *scheme = NULL;
 
+            SCMimeSmtpConfigExtractUrlsSchemeReset();
             TAILQ_FOREACH (scheme, &extract_urls_schemes->head, next) {
-                /* new_val_len: scheme value from config e.g. 'http' + '://' + null terminator */
-                size_t new_val_len = strlen(scheme->val) + 3 + 1;
-                if (new_val_len > UINT16_MAX) {
+                size_t scheme_len = strlen(scheme->val);
+                if (scheme_len > UINT16_MAX - SCHEME_SUFFIX_LEN) {
                     FatalError("Too long value for extract-urls-schemes");
                 }
-                char *new_val = SCMalloc(new_val_len);
-                if (unlikely(new_val == NULL)) {
-                    FatalError("SCMalloc failure.");
+                if (scheme->val[scheme_len - 1] != '/') {
+                    scheme_len += SCHEME_SUFFIX_LEN;
+                    char *new_val = SCMalloc(scheme_len);
+                    if (unlikely(new_val == NULL)) {
+                        FatalError("SCMalloc failure.");
+                    }
+                    int r = snprintf(new_val, scheme_len, "%s://", scheme->val);
+                    if (r != (int)scheme_len) {
+                        FatalError("snprintf failure.");
```

Comment:
need a better error message here, users will not know what happens

Reply:
Should it become a `BUG_ON` as standard lib `snprintf` should not fail ?